github-connector/routing_test.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/KarolJaksik/hack-showcase/github-connector/mocks"
+	"github.com/google/go-github/github"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type payload struct {
+	Anything string `json:"anything"`
+}
+
+//createRequest creates an HTTP request for test purposes
+func createRequest(t *testing.T) (*httptest.ResponseRecorder, *http.Request, *mocks.Validator, []byte) {
+
+	pld := payload{Anything: "test"}
+	toSend, err := json.Marshal(pld)
+	require.NoError(t, err)
+
+	req := httptest.NewRequest(http.MethodPost, "/webhook", bytes.NewBuffer(toSend))
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-Hub-Signature", "test")
+
+	rr := httptest.NewRecorder()
+
+	mockHandler := &mocks.Validator{}
+	mockPayload, err := json.Marshal(payload{Anything: "test"})
+	require.NoError(t, err)
+	return rr, req, mockHandler, mockPayload
+}
+
+func TestWebhookHandler_TestBadSecret(t *testing.T) {
+	t.Run("should respond with 401 status code", func(t *testing.T) {
+		// given
+
+		pld := payload{Anything: "test"}
+		toSend, err := json.Marshal(pld)
+		require.NoError(t, err)
+
+		req := httptest.NewRequest(http.MethodPost, "/webhook", bytes.NewBuffer(toSend))
+		req.Header.Set("Content-Type", "application/json")
+
+		rr := httptest.NewRecorder()
+		mockHandler := &mocks.Validator{}
+
+		mockHandler.On("GetToken").Return("test")
+		mockHandler.On("ValidatePayload", req, []byte("test")).Return(nil, errors.New("failed"))
+
+		// when
+		wh := NewWebHookHandler(mockHandler)
+
+		handler := http.HandlerFunc(wh.handleWebhook)
+		handler.ServeHTTP(rr, req)
+		// then
+		assert.Equal(t, http.StatusUnauthorized, rr.Code)
+
```

Comment:
Please add:
`mockHandler.AssertExpectations(t)`

This will make sure that all functions that were mocked by `mockHandler.On(...)` were called.